float32 measure_length
int32 measure_degree
